driver-core/src/main/java/com/datastax/driver/core/SchemaParser.java

```diff
@@ -735,4 +761,78 @@ private boolean references(DataType dataType, String typeName) {
             return false;
         }
     }
+
+    private static class V4SchemaParser extends V3SchemaParser {
+
+        private static final String SELECT_KEYSPACES = "SELECT * FROM system_schema.keyspaces";
+        private static final String SELECT_TABLES = "SELECT * FROM system_schema.tables";
+        private static final String SELECT_COLUMNS = "SELECT * FROM system_schema.columns";
+        private static final String SELECT_USERTYPES = "SELECT * FROM system_schema.types";
+        private static final String SELECT_FUNCTIONS = "SELECT * FROM system_schema.functions";
+        private static final String SELECT_AGGREGATES = "SELECT * FROM system_schema.aggregates";
+        private static final String SELECT_INDEXES = "SELECT * FROM system_schema.indexes";
+        private static final String SELECT_VIEWS = "SELECT * FROM system_schema.views";
+        private static final String SELECT_VIRTUAL_KEYSPACES = "SELECT * FROM system_virtual_schema.keyspaces";
+        private static final String SELECT_VIRTUAL_TABLES = "SELECT * FROM system_virtual_schema.tables";
+        private static final String SELECT_VIRTUAL_COLUMNS = "SELECT * FROM system_virtual_schema.columns";
+
+
+        private static final String TABLE_NAME = "table_name";
+
+        @Override
+        SystemRows fetchSystemRows(Cluster cluster, SchemaElement targetType, String targetKeyspace, String targetName, List<String> targetSignature, Connection connection, VersionNumber cassandraVersion)
+                throws ConnectionException, BusyConnectionException, ExecutionException, InterruptedException {
+
+            boolean isSchemaOrKeyspace = (targetType == null || targetType == KEYSPACE);
+
+            ResultSetFuture ksFuture = null,
```

Comment:
It would be nice to not duplicate a lot of this code from V3Parser, but refactoring it so it could be reused would probably be as much code, so i'm fine with this as is.

Reply:
That was my thought as well.